src/Microsoft.AspNetCore.Razor.Language/Syntax/TextSpan.cs

```diff
@@ -0,0 +1,261 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Internal;
+
+namespace Microsoft.AspNetCore.Razor.Language
+{
+    /// <summary>
+    /// Immutable abstract representation of a span of text.  For example, in an error diagnostic that reports a
+    /// location, it could come from a parsed string, text from a tool editor buffer, etc.
+    /// </summary>
+    internal readonly struct TextSpan : IEquatable<TextSpan>, IComparable<TextSpan>
```

Comment:
Should this even be used? Or do we ultimately want to move away from `SourceSpan` in favor of this? Thoughts?

Reply:
Why would we do that?

Reply:
Why would we do what? Not sure which of the two you are referring to.

Reply:
`SourceSpan` was based on this but we have too many dependent pieces that rely on `SourceSpan`. If there are valuable helper methods here that could be used on `SourceSpan` I'd say copy em over.

Reply:
I don't understand why this is here. Can we please just have one type for this?